akka-stream/src/main/scala/akka/stream/impl/FanIn.scala

```diff
@@ -151,6 +155,8 @@ private[akka] object FanIn {
 
     def isCancelled(input: Int): Boolean = cancelled(input)
 
+    def isAllCompleted(): Boolean = (0 to inputCount - 1).map(completed).forall(_ == true)
```

Comment:
OTOH I will probably remove this file completely in the upcoming days, so I am not sure how much optimization effort should be put into this, but the counter feels simple enough. We need the test cases though anyway.